system/assets/js/admin/presidecore/formFields.js

```diff
@@ -7,14 +8,56 @@
 	$(".auto-slug").each( function(){
 		var $this = $(this)
 		  , $basedOn = $this.parents("form:first").find("[name='" + $this.data( 'basedOn' ) + "']");
+		 
+		 if($this.val() != ''){
+		 	URLslug($this.val(),ulrPrefix);
+		 }
 
 		$basedOn.keyup( function(e){
 			var slug = $basedOn.val().replace( /\W/g, "-" ).replace( /-+/g, "-" ).toLowerCase();
-
 			$this.val( slug );
+			URLslug(slug,ulrPrefix);
+		} );
+		$this.keyup( function(e){
+			var slug = $this.val().replace( /\W/g, "-" ).replace( /-+/g, "-" ).toLowerCase();
+			$this.val( slug );
+			URLslug(slug,ulrPrefix);
 		} );
 	});
 
+	function URLslug(slug,prefix){
+		$(".auto-slug").nextAll("span").first().text('');
+		
+		if(slug == 'msg'){
+			$(".auto-slug").after("<span style='color:red;'>"+prefix+"</span>");
+			$(".auto-slug").attr('disabled', true);
+		} else if (slug != ''){
+			$(".auto-slug").attr('disabled', false);
+			$(".auto-slug").after("<span><b>URL slug</b>: http://"+prefix + slug+".html</span>");
```

Comment:
The protocol should sit within the prefix (a site can be set to use https so it shouldn't be hard coded here)